core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -77,21 +83,25 @@
   @Nullable
   private final Throttle throttle;
 
+  private Set<Substream> activeHedges; // not null and unchanged once isHedging = true
+  private ConcurrentRescheduler rescheduler; // not null and unchanged once isHedging = true
+
   private volatile State state = new State(
       new ArrayList<BufferEntry>(8), Collections.<Substream>emptyList(), null, false, false);
 
   /**
    * Either transparent retry happened or reached server's application logic.
    */
-  private boolean noMoreTransparentRetry;
+  private final AtomicBoolean noMoreTransparentRetry = new AtomicBoolean();
```

Comment:
I think this should be a property of the substream. It would 1) be simpler as there would not be any synchronization and 2) sort of seems more appropriate to allow one transparent retry _per attempt_. The spec isn't really clear how transparent retries mixes with normal retries and hedging. We should probably get that clarified.

Reply:
Let's get confirmation on that, because I was in the meeting where it was discussed and I'd expect it to be one per attempt. I expect most people don't care too much, but if it is an opportunity to remove synchronized state, that would be a win.

Reply:
Now transparent retry does not change the size of `activeHedges`, does not affect `hasPotentialHedging`.